service/history/replication/task_processor_manager.go

```diff
@@ -328,6 +343,36 @@ func (r *taskProcessorManagerImpl) cleanupReplicationTasks() error {
 	return err
 }
 
+func (r *taskProcessorManagerImpl) checkReplicationDLQSize() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	ctx = headers.SetCallerInfo(ctx, headers.SystemPreemptableCallerInfo)
+	defer cancel()
+	for clusterName := range r.shard.GetClusterMetadata().GetAllClusterInfo() {
+		if clusterName == r.shard.GetClusterMetadata().GetCurrentClusterName() {
+			continue
+		}
+
+		minTaskKey := r.shard.GetReplicatorDLQAckLevel(clusterName)
+		isEmpty, err := r.shard.GetExecutionManager().IsReplicationDLQEmpty(ctx, &persistence.GetReplicationTasksFromDLQRequest{
+			GetHistoryTasksRequest: persistence.GetHistoryTasksRequest{
+				ShardID:             r.shard.GetShardID(),
+				TaskCategory:        tasks.CategoryReplication,
+				InclusiveMinTaskKey: tasks.NewImmediateKey(minTaskKey),
+			},
+			SourceClusterName: clusterName,
+		})
+		if err != nil {
+			r.logger.Error("Failed to check replication DLQ size.", tag.Error(err))
+			return
+		}
+		if !isEmpty {
+			r.metricsHandler.Counter(metrics.ReplicationNonEmptyDLQCount.GetMetricName()).Record(1, metrics.OperationTag(metrics.ReplicationDLQStatsScope))
```

Comment:
I think gauge makes more sense to me in this case if we only want to know if a cluster has dlq messages or not. 

hmm actually I can also see the value of a counter, since it can give you a sense of how many shards are having dlq messages.